### DISCUSSION TITLE
Multi tenant Spring Repository

### DISCUSSION THREAD

**Discussion by Helferon** [flamingock/mongock#49] [General]
Using mongock multi tenant with Spring Repositories within a change unit, the repository uses different connection from MongoTemplate which is a bug because the data that queried and inserted goes to different db then intended

**Comment by dieppa** [comment]
Hello @Helferon ,

I understand your issue.  Springboot injects(doing its magic with dynamic proxies) the mongotemplate to the repositories, but each Mongock tenant has a different MongoTemplate, which is not the one injected to the repository. 

However, that's not really a bug. Take into account that Mongock professional is still on BETA and we haven't document it yet, but this won't be provided in version 1 neither. We'll try to address it in next version, but we cannot ensure that will be provided as we don't understand the implications yet.

As you can see in our [multi-tenant example](https://github.com/mongock/mongock-examples/tree/master/mongodb/springboot-multitenant) the workaround is to use directly MongoTemplate instead.


I hope this is helpful.

**Reply by Helferon** [reply]
Ok thanks for the quick reply @dieppa
